src/BugsnagUnity/Delivery.cs

```diff
@@ -301,6 +304,26 @@ IEnumerator PushToServer(IPayload payload, byte[] body)
                 _finishedCacheDeliveries.Add(payload.Id);
                 yield break;
             }
+
+            byte[] body = new byte[] { };
+            if (Application.platform == RuntimePlatform.WebGLPlayer)
+            {
+                body = PreaparePayloadBody(payload);
+            }
+            else
+            {
+                var bodyReady = false;
+                new Thread(() => {
+                    body = PreaparePayloadBody(payload);
+                    bodyReady = true;
+                }).Start();
+
+                while (!bodyReady)
+                {
+                    yield return null;
+                }
+            }
```

Comment:
body cannot be null as the c# compiler will complain that you are using something that might not ever be initialised, even though we know it will be. I also prefer using the bool for readability.